spec/requests/api/web/push_subscriptions_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'API Web Push Subscriptions' do
+  describe 'DELETE /api/web/push_subscriptions/:id' do
+    subject { delete api_web_push_subscription_path(token) }
+
+    let!(:web_push_subscription) do
+      Fabricate(:web_push_subscription)
+    end
+    let(:token) do
+      web_push_subscription.generate_token_for(:unsubscribe)
+    end
+
+    it 'deletes the subscription' do
+      expect { subject }
+        .to change(Web::PushSubscription, :count).by(-1)
+    end
```

Comment:
Probably needs a test case for the "subscription not found" case that renaud pointed out, just to assert it definitely always gives a 200 head response

Reply:
Good point. I just added two more test cases.

Reply:
This seems good to me now, I'll leave it up to y'all as to when/if this gets merged, but implementation wise this looks solid to me.